src/mongoc/mongoc-topology-scanner.c

```diff
@@ -398,13 +399,18 @@ mongoc_topology_scanner_start (mongoc_topology_scanner_t *ts,
 
    DL_FOREACH_SAFE (ts->nodes, node, tmp)
    {
-      if (mongoc_topology_scanner_node_setup (node)) {
+      /* FIXME: Gotta catch all the pokemons, not overwrite them */
+      bson_error_t error;
```

Comment:
Consider some logic something like this:

https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/topology.py#L369

The tests are here:

https://github.com/mongodb/mongo-python-driver/blob/master/test/test_topology.py#L593

You don't need to be quite as thorough and complex as PyMongo is, but some sort of intelligence about error messages would be great.